openedx/features/enterprise_support/api.py

```diff
@@ -233,52 +259,87 @@ def enterprise_enabled():
+        # and set the sso_slug parameter to match if so.
+        sso_slug = Registry.get_from_pipeline(running_pipeline).provider_id
 
-    if tpa_hint:
+    if sso_slug:
         # If we have a third-party auth provider, get the linked enterprise customer.
         try:
-            ec = EnterpriseCustomer.objects.get(enterprise_customer_identity_provider__provider_id=tpa_hint)
+            # FIXME: Implement an Enterprise API endpoint where we can get the EC
+            # directly via the linked SSO provider
+            ec_uuid = EnterpriseCustomer.objects.get(
+                enterprise_customer_identity_provider__provider_id=sso_slug
+            ).uuid
         except EnterpriseCustomer.DoesNotExist:
-            pass
+            ec_uuid = None
+    else:
+        ec_uuid = request.GET.get('enterprise_customer') or request.COOKIES.get(settings.ENTERPRISE_CUSTOMER_COOKIE_NAME)
 
-    ec_uuid = request.GET.get('enterprise_customer') or request.COOKIES.get(settings.ENTERPRISE_CUSTOMER_COOKIE_NAME)
+    if not ec_uuid and request.user.is_authenticated():
+        learner_data = get_enterprise_learner_data(request.site, request.user)
+        if learner_data:
+            ec_uuid = learner_data[0]['enterprise_customer']['uuid']
     # If we haven't obtained an EnterpriseCustomer through the other methods, check the
     # session cookies and URL parameters for an explicitly-passed EnterpriseCustomer.
-    if not ec and ec_uuid:
+    if ec_uuid:
         try:
-            ec = EnterpriseCustomer.objects.get(uuid=ec_uuid)
-        except (EnterpriseCustomer.DoesNotExist, ValueError):
+            ec = EnterpriseApiClient().get_enterprise_customer(ec_uuid)
+        except HttpNotFoundError:
             ec = None
 
     return ec
 
 
-def consent_needed_for_course(user, course_id):
+def consent_needed_for_course(request, user, course_id, enrollment_exists=False):
     """
     Wrap the enterprise app check to determine if the user needs to grant
     data sharing permissions before accessing a course.
     """
     if not enterprise_enabled():
         return False
-    return consent_necessary_for_course(user, course_id)
 
+    consent_key = ('data_sharing_consent_needed', course_id)
+
+    if request.session.get(consent_key) is False:
+        return False
 
-def get_enterprise_consent_url(request, course_id, user=None, return_to=None):
+    enterprise_learner_details = get_enterprise_learner_data(request.site, user)
+    if not enterprise_learner_details:
+        consent_needed = False
+    else:
+        client = ConsentApiClient()
+        consent_needed = any(
+            client.consent_needed(
+                username=user.username,
+                course_id=course_id,
+                enterprise_customer_uuid=learner['enterprise_customer']['uuid'],
+                enrollment_exists=enrollment_exists,
+            )
+            for learner in enterprise_learner_details
```

Comment:
Right now, due to the restrictions we have in place, there should only be one item.

The API call handles it in that it performs checks with the course ID against each enterprise's linked catalog.